### PR TITLE
Improve the agent management UX

### PR DESCRIPTION
**Pull request by sabaimran**
- Add support for seeing all steps of the agent modification flow via tabs at the top of the modal
- Separate knowledge base & tool selection into two separate parts

### CODE REVIEW

File: src/interface/web/components/ui/tabs.tsx
Comment (debanjum) [inline src/interface/web/components/ui/tabs.tsx:1]: If all required fields aren't populated the save fails but i don't know which required fields I didn't fill. need to go through each tab to check that. Maybe good to add some something in the tab names to indicate that the fields in that tab need to be filled/fixed (e.g a `*` suffix so Customize => Customize *) or the tab becomes red

File: src/interface/web/app/components/agentCard/agentCard.tsx
Comment (debanjum) [inline src/interface/web/app/components/agentCard/agentCard.tsx:1]: Not sure if we want to redirect not logged in users to sign-in before trying to edit a card. Right now I can fill all fields in an agent card but can't create it until logged in. This maybe annoying

**Review by debanjum — APPROVED** [review]
The tab changes look good as give better orientation of what all needs to be filled to create an agent instead of the previous Previous > Next with no breadcrumb like help

File: src/interface/web/components/ui/tabs.tsx
Comment (sabaimran) [reply src/interface/web/components/ui/tabs.tsx:1]: Nice feedback! Included both in #1015

**Pull request merged**